### PR TITLE
Update Markdown component guidance

### PR DESCRIPTION
**Pull request by FredKSchott**
- Added Markdown component warnings to the relevant sections
- Removed an outdated piece of docs around markdown customization (no longer possible in v1.0-beta)

### CODE REVIEW

**Comment by sarah11918** [comment]
Waiting on PR #405 (close to done) before this one!

File: src/pages/en/guides/markdown-content.md [hunk 1]
```diff
@@ -203,7 +203,7 @@ back to it later.
 
 ## Markdown Component
 
-Astro has a dedicated component used to let you render Markdown in `.astro` files. 
+> NOTE: The `<Markdown>` component does not work in SSR and may be removed before v1.0. It should should be avoided if possible. To use Markdown in your templates, use a seperate `.md` file and then `import` it into your template as a component. Read this [RFC Discussion](https://github.com/withastro/rfcs/discussions/179) to learn more.
```
Comment (sarah11918) [inline src/pages/en/guides/markdown-content.md:206]:
```suggestion
> NOTE: The `<Markdown>` component does not work in SSR and may be removed before v1.0. It should should be avoided if possible. To use Markdown in your templates, use a seperate `.md` file and then [`import` Markdown](/en/guides/markdown-content#importing-markdown) into your template as a component.
```
You could do something like this, now that #405 is merged.

**Comment by sarah11918** [comment]
@FredKSchott This will have conflicts to resolve now that #405 is merged. And, a reminder in my comment above that we can now link directly to info about importing Markdown in the docs, but otherwise, LGTM!

File: src/pages/en/guides/markdown-content.md (hunk 1 above)
Comment (sarah11918) [reply src/pages/en/guides/markdown-content.md:206]: (Unless your intention really was to point to the Discussion/rationale, of course! But, I do think having the link to how to import Markdown content, now that we have it, works here.)

Comment (FredKSchott) [reply src/pages/en/guides/markdown-content.md:206]: Agreed! That was a bit of a placeholder

**Comment by FredKSchott** [comment]
Thanks @sarah11918 !

**Pull request merged**